python/ql/lib/semmle/python/regex.qll

```diff
@@ -344,9 +343,8 @@ abstract class RegexString extends Expr {
     this.escapingChar(start) and
     this.getChar(start + 1) = "N" and
     this.getChar(start + 2) = "{" and
-    this.getChar(end - 1) = "}" and
     end > start and
-    not exists(int i | start + 2 < i and i < end - 1 | this.getChar(i) = "}")
+    end - 1 = min(int i | start + 2 < i and this.getChar(i) = "}")
```

Comment:
Here we can also delete `end > start` since we know that `end - 1 > start + 2`.

Reply:
Slightly _less_ immediate, perhaps, but I've done this as well. :slightly_smiling_face: